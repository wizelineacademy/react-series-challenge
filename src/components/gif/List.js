import React from 'react';
import Item from './Item';

const List = ({ loading, gifs, favorites, toggleFavorite }) => {
  if (loading) {
    return <div>Loading trending gifs!</div>;
  }

  return Object.keys(gifs).map(id => {
    const gif = gifs[id];
    return (
      <Item
        key={id}
        data={gif}
        toggleFavorite={toggleFavorite}
        isFavorite={!!favorites[gif.id]}
      />
    );
  });
};

export default List;
