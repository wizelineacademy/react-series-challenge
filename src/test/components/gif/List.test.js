import React from 'react';
import { shallow, mount, render } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import List from '../../../components/gif/List';
import configureStore from 'redux-mock-store';

const middlewares = [];
const mockStore = configureStore(middlewares);

describe('Favorites component', () => {
  test('should render correctly that is loading if i passed loading=true', () => {
    const component = shallow(<List loading={true} />);

    expect(shallowToJson(component)).toMatchSnapshot();
  });

  test('should render correctly that is loading if i passed loading=false', () => {
    const component = shallow(<List loading={false} gifs={{}} />);

    expect(shallowToJson(component)).toMatchSnapshot();
  });

  test('should be able to render gifs', () => {
    const fakeGifs = {
      1: {
        title: 'best meme'
      }
    };

    const component = shallow(
      <List loading={false} gifs={fakeGifs} favorites={{}} />
    );

    expect(shallowToJson(component)).toMatchSnapshot();
  });
});
