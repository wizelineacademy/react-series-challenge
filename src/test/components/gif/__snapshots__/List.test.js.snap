// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Favorites component should be able to render gifs 1`] = `
<styled.div>
  <Component
    data={
      Object {
        "title": "best meme",
      }
    }
    isFavorite={false}
    key="1"
  />
</styled.div>
`;

exports[`Favorites component should render correctly that is loading if i passed loading=false 1`] = `<styled.div />`;

exports[`Favorites component should render correctly that is loading if i passed loading=true 1`] = `
<div>
  Loading trending gifs!
</div>
`;
